networking_calico/agent/dhcp_agent.py

```diff
@@ -501,7 +501,7 @@ def _pre_snapshot_hook(self):
         # Dnsmasqs if no longer needed.  Also remove all port and subnet
         # information.
         LOG.debug("Reset cache for new snapshot")
-        for network_id in self.agent.cache.get_network_ids():
+        for network_id in list(self.agent.cache.get_network_ids()):
```

Comment:
@lmm Curious why PR is now closed.  Do we not still need **this** change?

Reply:
@neiljerram It doesn't seem like it. In Python 2, `keys()` returns a copy and so doesn't hit the issue as with Python 3.

> keys()
> Return a copy of the dictionary’s list of keys. See the note for dict.items().

https://docs.python.org/2/library/stdtypes.html#dict.keys

And here's Python 3's doc on `keys()`:

> keys()
> Return a new view of the dictionary’s keys. See the documentation of view objects.

https://docs.python.org/3/library/stdtypes.html#dict.keys

Reply:
I'm sorry, I was being stupid.  IIRC this change is needed by the move from Python 3.7 to 3.8, and I somehow forgot that we are using Python 2 in this branch.